bigflow/bigquery/dataset_configuration.py

```diff
@@ -67,38 +66,16 @@ def add_configuration(self,
         if dataset_labels:
             all_properties['dataset_labels'] = dataset_labels
 
-        if credentials:
-            # all_properties['impersonate_service_account'] = self._credentials_for_impersonate_service_account(
-            #     impersonate_service_account)
-
-            all_properties['credentials'] = credentials
-
-        logger.info("0o0o0o0o0o0o0o0o0o0o0o0o0o0 all_properties", all_properties)
-        logger.info("0o0o0o0o0o0o0o0o0o0o0o0o0o0 env", env)
-
+        if impersonate_service_account:
+            all_properties['impersonate_service_account'] = impersonate_service_account
 
         self.delegate.add_configuration(env, all_properties, is_default=is_default)
         return self
 
     def create_dataset_manager(self, env: str = None) -> Dataset:
-        logger.info("0o0o0o0o0o0o0o0o0o0o0o0o0o0 InteractiveDatasetManager in create_dataset_manager",
-                    InteractiveDatasetManager(
-                        project_id=self.resolve_project_id(env),
-                        dataset_name=self.resolve_dataset_name(env),
-                        internal_tables=self.resolve_internal_tables(env),
-                        external_tables=self.resolve_external_tables(env),
-                        extras=self.resolve_extra_properties(env),
-                        tables_labels=self.resolve_tables_labels(env),
-                        dataset_labels=self.resolve_dataset_labels(env),
-                        # credentials=self.credentials
-                        credentials=self.resolve_credentials(env)
-                    ).__dict__
-                    )
-
-        # if passed_credentials is None:
-        #     creds = self.credentials
-        # else:
-        #     creds = passed_credentials
+
+        resolved_impersonate_service_account = self.resolve_impersonate_service_account(env)
+        resolved_credentials = self._credentials_for_impersonate_service_account(resolved_impersonate_service_account)
```

Comment:
I don't really get those two lines. Could you please elaborate on what does the first resolution do?

Reply:
I wanted to pass in DatasetConfig constructor impersonate_service_account so it would be the same as in config responsible for Dataflow workflow. InteractiveDatasetManager requires credentials as google.auth.credentials.Credentials so in 77 I resolve impersonate_service_account from config for given env (same as every other field in DatasetConfig's constructor) and in 78 I create credentials (type google.auth.credentials.Credentials) out of SA for specific environment.